src/App.jsx

```diff
@@ -14,45 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'alphabetically';
+const SORT_BY_LENGTH = 'length';
+
+function getPreparedGoods(goods, isReversed, sortType) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((a, b) => {
+      switch (sortType) {
+        case SORT_ALPHABETICALLY:
+          return a.localeCompare(b);
+
+        case SORT_BY_LENGTH:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [theWayToSort, setTheWayToSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedList = getPreparedGoods(
+    goodsFromServer,
+    isReversed,
+    theWayToSort,
+  );
+  const resetChanges = () => {
+    setTheWayToSort('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setTheWayToSort(SORT_ALPHABETICALLY)}
+          type="button"
+          className={`button is-info ${theWayToSort !== SORT_ALPHABETICALLY && ('is-light')}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setTheWayToSort(SORT_BY_LENGTH)}
+          type="button"
+          className={`button is-success ${theWayToSort !== SORT_BY_LENGTH && ('is-light')}`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+          type="button"
+          className={`button is-warning ${!isReversed && ('is-light')}`}
+        >
+          Reverse
+        </button>
+
+        {(sortedList.toString() !== goodsFromServer.toString()
```

Comment:
Such check is a little bit unreliable, better to check if some type of sort is selected